### PR TITLE
Generate PDFs with chromium, remove wicked_pdf

### PR DESCRIPTION
**Pull request by wioux**
wkhtmltopdf has more limitations and bugs (links in our PDFs are not clickable!), and we also have issues with having to keep an xvfb display running in our container.

As mentioned in #447 ideally we would have chrome running in a separate container. Right now the `chrome-headless-render-pdf` package I'm using doesn't support connecting to a remote instance but it looks like an easy-ish patch we could make. I intend to follow up on it but meanwhile I think we are somewhat better off running chrome headless rather than xvfb + wicked_pdf.

Depends on #450

### CODE REVIEW

File: app/assets/stylesheets/pdf.scss [hunk 1]
```diff
@@ -195,3 +195,37 @@ h1 {
...
+}
+
+#header img {
+  width: 100%;
+}
+
+#main-content {
+  margin-top: 65px;
+}
+
```
Comment (k-stewart) [inline app/assets/stylesheets/pdf.scss]: nit: extra blank line

File: lib/tasks/pdf.rake [hunk 2]
```diff
@@ -4,7 +4,6 @@ namespace :pdfs do
   task recreate: :environment do
     Lesson.published.find_each do |lesson|
       UpdateLessonPdf.perform_now(lesson.id)
-      sleep 0.5
```
Comment (k-stewart) [inline lib/tasks/pdf.rake]: 🎉

File: spec/lib/pdf_template_spec.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                      at_least(:once)
+      expect(ENV).to receive(:[]).with("SERVER_PROTOCOL").
+                      and_return(proto).
+                      at_least(:once)
+      expect(ENV).to receive(:[]).with("SERVER_PORT").
+                      and_return(nil).
+                      at_least(:once)
+
+      html = %(<a href="/relative">anchor</a>")
+      expect(pdf_template.send(:rebase_urls, html)).
```
Comment (k-stewart) [inline spec/lib/pdf_template_spec.rb]: why is it necessary to send `:rebase_urls`?

Comment (wioux) [reply spec/lib/pdf_template_spec.rb]: It's a private method.

Comment (k-stewart) [reply spec/lib/pdf_template_spec.rb]: In that case, maybe consider finding some other way to test?  One of the only purposes of making a method private is to tell coders that it's not a reliable feature of the object's public interface.  [The internet is divided](https://www.google.com/search?q=why+not+test+private+methods) over this, but that's how I was taught.

**Comment by wioux** [comment]
@vbrown608 For now I'd like to take the `--no-sandbox` approach and add a seccomp profile later on if/when we move the chromium process to a separate container.

File: lib/pdf_template.rb [hunk 4]
```diff
@@ -1,41 +1,70 @@
...
+
+    print_pdf(input.path)
+  end
+
+  private
+
+  def print_pdf(input)
+    output = Tempfile.new(["pdf", ".pdf"])
+    chrome_wrapper = Rails.root.join("bin/chrome-stub").to_s
+    command = ["timeout", "-t", "5",
```
Comment (k-stewart) [inline lib/pdf_template.rb]: Pls use Ruby's timeout instead, for cross-platform compatibility!

File: Dockerfile [hunk 5]
```diff
@@ -28,6 +29,7 @@ RUN echo "@edge http://nl.alpinelinux.org/alpine/edge/main" >>/etc/apk/repositor
```
Comment (vbrown608) [inline Dockerfile]: Can we remove xvfb and wkhtmltopdf@edgetesting from the Dockerfile?

File: app/assets/stylesheets/pdf.scss [hunk 6]
```diff
@@ -196,6 +195,25 @@ h1 {
...
+  right: 0;
+}
+
+#header img {
+  width: 100%;
+}
+
+#main-content {
+  margin-top: 65px;
+}
```
Comment (vbrown608) [inline app/assets/stylesheets/pdf.scss]: Some styles aren't getting picked up when I test this locally and the image also doesn't seem to be coming through. I'm not sure if this is unique to my setup, maybe we can check in about it on Monday.

pdf

Comment (wioux) [reply app/assets/stylesheets/pdf.scss]: I pushed a change to inline the stylesheet and use a data URI for the header image, which should be more reliable.

File: Dockerfile (hunk 5 above)
Comment (wioux) [reply Dockerfile]: Yep! Good idea.

**Comment by wioux** [comment]
This is ready for review again. I switched to a different node library, which doesn't have the problem with stalling jobs (though I'm keeping it wrapped in a timeout block just in case). It also has better support for printing from an existing/remote chrome process (instead of spawning a new one each time) so we should be able to follow up soon with moving the browser to its own container.

**Pull request merged**